### ISSUE TITLE
Can't add a subfolder of a root folder as another root folder

### ISSUE THREAD

**Issue by nikolaydyankov** [atom/atom#5662]
Example:

Folder_A
----- Folder_B
----- Folder_C
----- Folder_D
---------- Folder_E
---------- Folder_F

In this situation Folder_D can't be added as a root folder. Desired result:

Folder_A
----- Folder_B
----- Folder_C
----- Folder_D
---------- Folder_E
---------- Folder_F
Folder_D
----- Folder_E
----- Folder_F

**Comment by karloluis** [comment]
Well you ARE trying to access already locked memory space. This would be conflicting to keep files uncorrupt upon saves. Is there anything specific that you wish to perform? It seems there would be no need for this level of functionality since the folder is already in the scope of Atom's process.

**Comment by nikolaydyankov** [comment]
For example I have my "Work" folder as my root folder and I have a WordPress installation within that folder. I would like to add the "wp-plugins" as a second Root Folder for easy access.

You could do that in Sublime and I found it very convenient.

**Comment by batjko** [comment]
As far as I know multiple root folders are not supported yet.
It's an often-requested feature and on the list though.

**Comment by nikolaydyankov** [comment]
It's in version 0.182.0

**Comment by karloluis** [comment]
I was about to write something along the lines of what @batjko just said but it seems that there is such a functionality. I figured what you ment by this post and wil be coming up with something shortly. In the meanwhile check out the v1.0 goals because there might be some information about this on there.

**Comment by batjko** [comment]
Ah! There is an "Add Root folder" feature now.
Wow, that completely passed me by.

**Comment by nikolaydyankov** [comment]
Sorry @karloluis I didn't understand... "wil be coming up with something shortly" - so this feature will be added soon?

**Comment by karloluis** [comment]
It seems there is a mechanism to this, it is able to root parent directories but not sub's. I do not know for sure but I am fairly certain it is a safety method. Managing the scope of the accessed files is tricky when you have multiple instances on callback.

Think of it like this. 
Folder_A
----- Folder_B
----- Folder_C
----- Folder_D
---------- Folder_E
---------- Folder_F

If you where to open A first and then F, F is a child therefore there is no way he can tell that A contains because of scope mechanisms, the F instance does not know that A has access to the data and conflicts arise unavoidably.
Hence if you opened F and then A, A is capable of identifying his children therefore permitted. No conflict would arise from this because the scope has been broadened.

**TL;DR** Open the sub-folders you want to use first and work your way up as necessary. Use this wisely, root folders that provide a useful scope of the current work-space you need.

Hope this helps.

**Comment by nikolaydyankov** [comment]
Thanks, that worked!

**Comment by karloluis** [comment]
Glad to help.
